src/logger.ts

```diff
@@ -12,36 +12,69 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import * as consoleLogLevel from 'console-log-level';
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
 import {defaultConfig} from './config';
+import {LogSync, Logging} from '@google-cloud/logging-min';
+
+const logging = new Logging();
+
+logging.setProjectId().catch(err => {
+  console.error(`failed to set logging project id ${err}`);
+});
+logging.setDetectedResource().catch(err => {
+  console.error(`failed to discover resource metadata ${err}`);
+});
 
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const pjson = require('../../package.json');
 
-const LEVEL_NAMES: consoleLogLevel.LogLevelNames[] = [
-  'fatal',
-  'error',
-  'warn',
-  'info',
-  'debug',
-  'trace',
-];
-
-function logLevelToName(level?: number): consoleLogLevel.LogLevelNames {
-  if (level === undefined) {
-    level = defaultConfig.logLevel;
-  } else if (level < 0) {
-    level = 0;
-  } else if (level > 4) {
-    level = 4;
+export class Logger {
+  private log: LogSync;
+  private severityThreshold: number;
+
+  constructor(readonly level?: number) {
+    if (level === undefined) {
+      level = defaultConfig.logLevel;
+    }
+    if (level < 0) {
+      level = 0;
+    } else if (level > 4) {
+      level = 4;
+    }
+    this.severityThreshold = level;
+    this.log = logging.logSync(pjson.name);
+  }
+
+  debug(msg: string) {
+    if (this.severityThreshold > 3) {
+      this.log.debug(this.log.entry(this.toOneLine(msg)));
+    }
+  }
+
+  info(msg: string) {
+    if (this.severityThreshold > 2) {
+      this.log.info(this.log.entry(this.toOneLine(msg)));
+    }
+  }
+
+  warn(msg: string) {
+    if (this.severityThreshold > 1) {
+      this.log.warning(this.log.entry(this.toOneLine(msg)));
+    }
+  }
+
+  error(msg: string) {
+    if (this.severityThreshold > 0) {
+      this.log.error(this.log.entry(this.toOneLine(msg)));
+    }
+  }
+
+  private toOneLine(msg: string): string {
+    return msg.replace('\n', '\\n');
```

Comment:
Do you need to handle '\r' as well?

Reply:
I do not think that standalone `\r` should be handled. Would the following handle Windows new lines IYO:
```java
var temp = msg.replace('\r\n', '\\r\\n');
return temp.replace('\n', '\\n');
```

Reply:
I saw it a common practice of capturing multiline payload in Json is to replace with an anchor that can be used to pretty print the text later.
It is hard to restore the original multi-line string later (e.g. callstacks). Such work will require from a reader a lot of effort and manual work. With the anchor, a simple find-and-replace in an editor or with a script/code will do the work.
Would removing \r means that in certain OS (e.g. Windows) restoring will not work?

Reply:
Just updated the code. Please, see the new version that handles new lines in both Window and Linux formats.